template/README.md

```diff
@@ -13,30 +13,128 @@ Documentation on [parent repository](https://github.com/fredrikhgrelland/vagrant
+1. Create a [Nomad job specification](https://www.nomadproject.io/docs/job-specification). This should be placed in [conf/nomad/](conf/nomad/README.md). [Example hcl-file](test_example/conf/nomad/countdash.hcl)
+1. Use terraform to deploy this job to Nomad. Refer to ["Terraform module requirements"](#terraform-module-requirements) on how to do this. [Example](test_example/example)
+
+## File structure
+```text
+|-- conf/
+|     |-- nomad/        # nomad job specifications
+|
+|-- dev/                # files used to control testing and development
+|   |-- ansible/        # playbooks run at startup
+|   |-- vagrant/        # files altering the vagrant-box
+|       |-- conf/
+|
+|-- docker/             # files used to build docker-image
+|
+|-- example/            # terraform-files using the terraform-module
+|
+|-- test_example/       # a copy of template, filled with example-code
+```
+
+## Terraform module requirements
+[Hashiscorp documentation](https://www.terraform.io/docs/modules/index.html)
+
+ This template follows Hashicorp's standards for terraform modules. In short the root should contain a `main.tf`, `variables.tf` and `outputs.tf`. These three form a module. In addition to this there should be an `example/` folder that uses the module. To get a feel for how to use this effectively in our case you can look at the files [main.tf](test_example/main.tf), [variables.tf](test_example/variables.tf), and [outputs.tf](test_example/outputs.tf) from [test_example/](test_example/).
+
+ Project structure:
+ ```text
+|-- main.tf
+|-- variables.tf
+|-- outputs.tf
+|
+|-- example/
+|   |-- main.tf
+|   |-- variables.tf
+|   |-- outputs.tf
+```
+
+## Using the template and box
+
+### Commands
+There are several commands that help to run the vagrant-box:
+- `make install` installs all prerequisites
+
+- `make up` provisions a [vagrant-hashistack](https://github.com/fredrikhgrelland/vagrant-hashistack/) box on your machine. After the machine and hashistack are set up it will run the default [startup scheme](#default-startup-scheme).
+
+- `make clean` take down the provisioned box if there is any.
+
+- `make update` downloads the newest version of the [vagrant-hashistack box](https://github.com/fredrikhgrelland/vagrant-hashistack/) from [vagrantcloud](https://vagrantcloud.com/fredrikhgrelland/hashistack).
+
+- `make example` runs the example in [test_example/](test_example/)
+
+For full info, check `template/Makefile`
+
+`!Note` - Makefile commands are not idempotent in the context of vagrant-box.  You could face the error of port collisions. Most of the cases it could happen because of the vagrant box has already been running.
+
+Once vagrant-box is running, you can use other options like the Nomad- and Terraform-CLIs to iterate over the deployment in the development stage.
 
```

Comment:
```suggestion
### Iterration of development process
Once you start the box with one of the command `make dev`, `make up` or `make example`, 
you need simple way how to continiously deploy development changes.

There are several options.

1. You can install Hashicorp binaries on local machine, such as terraform and nomad.
Than you can deploy chnages to the vagrant-box using these binaries.

Example terraform:
\```
# From the local machine
terraform init
terraform apply
\```

Example nomad:
\```
nomad job run countdash.hcl
\```

2. Box instance has all binaries are installed and available in the PATH.
You can use ´vagrant ssh´ to call commands

\´´´
# remote 
vagrant ssh default -c 'cd /vagrant; terraform init; terraform apply'

# local
vagrant ssh default
cd /vagrant
terraform init
terraform apply
\´´´

Note: default is the name of running VM. You could also use VM `id`.
To get vm `id` check `vagrant global-status`.
```

Reply:
Just remove `\` for 99, 103, 106, 108, 113, 122

Reply:
We migrate this to another PR. RESOLVED for now